src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/Common/SafeNativeMethods.cs

```diff
@@ -143,7 +143,7 @@ sealed internal class Wrapper
             [ResourceConsumption(ResourceScope.Process, ResourceScope.Process)]
             static internal void ClearErrorInfo()
             { // MDAC 68199
-                SafeNativeMethods.SetErrorInfo(0, ADP.PtrZero);
+                SafeNativeMethods.SetErrorInfo(0, ADP.s_ptrZero);
```

Comment:
Same as empty string, there's really no point in using a reference to a variable which is zero instead of just using the literal zero and letting the compiler and jit see it as a const.

Reply:
If we could use net6 then we'd want to use `(nint)0` but we can't so either `new IntPtr(0)` or `IntPtr.Zero` so we don't need our own constant.

Reply:
I obsess about this change and [taking out the Win9x assessment](https://github.com/dotnet/SqlClient/pull/1306#discussion_r719870325), and I'm not sure if it's a good idea to include them in this PR if they already have been approved.
cc @cheenamalhotra

Reply:
The objective is to merge. If it's not clear that changing to use IntPtr or taking out those assertions is correct at this time then I'm fine with leaving them in and coming back later, they aren't incorrect or causing a problem.